modules/azure/client_factory.go

```diff
@@ -0,0 +1,147 @@
+/*
+
+This file implements an Azure client factory that automatically handles setting up Base URI
+values for sovereign cloud support. Note the list of clients below is not initially exhaustive;
+rather, additional clients will me added as-needed.
+
+*/
+
+package azure
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute"
+	"github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2019-11-01/containerservice"
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2019-06-01/subscriptions"
+	autorestAzure "github.com/Azure/go-autorest/autorest/azure"
+)
+
+const (
+	// AzureEnvironmentEnvName is the name of the Azure environment to use. Set to one of the following:
+	//
+	// "AzureChinaCloud":        ChinaCloud
+	// "AzureGermanCloud":       GermanCloud
+	// "AzurePublicCloud":       PublicCloud
+	// "AzureUSGovernmentCloud": USGovernmentCloud
+	// "AzureStackCloud":		 Azure stack
```

Comment:
Should these be available in exported `const` vars?

Reply:
The environment vars would be set outside any code - listed here for reference only.